aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -108,6 +109,11 @@ namespace detail {
   }
 }
 
+using DispatchKeySetFuncType = std::function<DispatchKeySet(DispatchKeySet)>;
+
+TORCH_API ska::flat_hash_map<DispatchKey, DispatchKeySetFuncType>&
+getDKFuncMaps();
```

Comment:
This is probably going to be the most controversial part of this patch. Let me first see if I can convince myself that there is not an easier way to do it.